test/TableGateway/Feature/FeatureSetTest.php

```diff
@@ -77,4 +78,93 @@ public function testAddFeatureThatFeatureHasTableGatewayButFeatureSetDoesnotHas(
     }
+
+    /**
+     * @cover FeatureSet::canCallMagicCall
+     */
+    public function testCanCallMagicCallReturnsTrueForAddedMethodOfAddedFeature()
+    {
+        $feature = new SequenceFeature('id', 'table_sequence');
+        $featureSet = new FeatureSet;
+        $featureSet->addFeature($feature);
+
+        $this->assertTrue(
+            $featureSet->canCallMagicCall('lastSequenceId'),
+            "Should have been able to call lastSequenceId from the Sequence Feature"
+        );
+    }
+
+    /**
+     * @cover FeatureSet::canCallMagicCall
+     */
+    public function testCanCallMagicCallReturnsFalseForAddedMethodOfAddedFeature()
+    {
+        $feature = new SequenceFeature('id', 'table_sequence');
+        $featureSet = new FeatureSet;
+        $featureSet->addFeature($feature);
+
+        $this->assertFalse(
+            $featureSet->canCallMagicCall('postInitialize'),
+            "Should have been able to call postInitialize from the MetaData Feature"
+        );
+    }
+
+    /**
+     * @cover FeatureSet::canCallMagicCall
+     */
+    public function testCanCallMagicCallReturnsFalseWhenNoFeaturesHaveBeenAdded()
+    {
+        $featureSet = new FeatureSet;
+        $this->assertFalse(
+            $featureSet->canCallMagicCall('lastSequenceId')
+        );
+    }
+
+    /**
+     * @cover FeatureSet::callMagicCall
+     */
+    public function testCallMagicCallSucceedsForValidMethodOfAddedFeature()
+    {
+        $sequenceName = 'table_sequence';
+
+        $platformMock = $this->getMock('Zend\Db\Adapter\Platform\Postgresql');
+        $platformMock->expects($this->any())
+            ->method('getName')->will($this->returnValue('PostgreSQL'));
+
+        $resultMock = $this->getMock('Zend\Db\Adapter\Driver\Pgsql\Result');
+        $resultMock->expects($this->any())
+            ->method('current')
+            ->will($this->returnValue(['currval' => 1]));
+
+        $statementMock = $this->getMock('Zend\Db\Adapter\Driver\StatementInterface');
+        $statementMock->expects($this->any())
+            ->method('prepare')
+            ->with('SELECT CURRVAL(\'' . $sequenceName . '\')');
+        $statementMock->expects($this->any())
+            ->method('execute')
+            ->will($this->returnValue($resultMock));
+
+        $adapterMock = $this->getMockBuilder('Zend\Db\Adapter\Adapter')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $adapterMock->expects($this->any())
+            ->method('getPlatform')->will($this->returnValue($platformMock));
+        $adapterMock->expects($this->any())
+            ->method('createStatement')->will($this->returnValue($statementMock));
+
+        $tableGatewayMock = $this->getMockBuilder('Zend\Db\TableGateway\AbstractTableGateway')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $reflectionClass = new \ReflectionClass('Zend\Db\TableGateway\AbstractTableGateway');
```

Comment:
you may use ReflectionProperty instead, also, move to use statement instead of adding "\" in front of it.